src/vs/workbench/contrib/chat/browser/chatContentParts/toolInvocationParts/chatTerminalToolProgressPart.ts

```diff
@@ -58,6 +58,11 @@ const sanitizerConfig = Object.freeze<DomSanitizerConfig>({
 
 let lastFocusedProgressPart: ChatTerminalToolProgressPart | undefined;
 
+/**
+ * Remembers whether a tool invocation was last expanded so state survives virtualization re-renders.
+ */
+const expandedStateByInvocation = new WeakMap<IChatToolInvocation | IChatToolInvocationSerialized, boolean>();
```

Comment:
If that's the reason it's collapsing on scroll, we should serialize it properly, then it'll work on scroll and reload I think? @roblourens @justschen any tips?

Reply:
You mean adding it to `IChatTerminalToolInvocationData`? I thought we only wanted to touch that in the tool